mesa/space.py

```diff
@@ -81,237 +62,207 @@ def __init__(self, width, height, torus):
-
-    def get_neighborhood(self, pos, moore,
-                         include_center=False, radius=1):
-        """ Return a list of cells that are in the neighborhood of a
-        certain point.
-
-        Args:
-            pos: Coordinate tuple for the neighborhood to get.
-            moore: If True, return Moore neighborhood
-                   (including diagonals)
-                   If False, return Von Neumann neighborhood
-                   (exclude diagonals)
-            include_center: If True, return the (x, y) cell as well.
-                            Otherwise, return surrounding cells only.
-            radius: radius, in cells, of neighborhood to get.
+            A set of adjacent cells of a single cell at `pos`. 
 
-        Returns:
-            A list of coordinate tuples representing the neighborhood;
-            With radius 1, at most 9 if Moore, 5 if Von Neumann (8 and 4
-            if not including the center).
+            The number of cells 
+            in the Moore neighborhood with radius n is (2n+1)^2 -1.
+            The number of cells in the Moore neighborhood with radius n is [(2n+1)^2 -1]
+            (http://www.conwaylife.com/wiki/Moore_neighbourhood).
 
+            The number of cells in the von Neumann neighbourhood of 
+            radius n of a single cell is 
+            [2n(n+1)](http://www.conwaylife.com/wiki/Von_Neumann_neighborhood).
         """
-        return list(self.iter_neighborhood(pos, moore, include_center, radius))
-
-    def iter_neighbors(self, pos, moore,
-                       include_center=False, radius=1):
-        """ Return an iterator over neighbors to a certain point.
-
-        Args:
-            pos: Coordinates for the neighborhood to get.
-            moore: If True, return Moore neighborhood
-                    (including diagonals)
-                   If False, return Von Neumann neighborhood
-                     (exclude diagonals)
-            include_center: If True, return the (x, y) cell as well.
-                            Otherwise,
-                            return surrounding cells only.
-            radius: radius, in cells, of neighborhood to get.
-
-        Returns:
-            An iterator of non-None objects in the given neighborhood;
-            at most 9 if Moore, 5 if Von-Neumann
-            (8 and 4 if not including the center).
+        if moore:
+            neighbors = self._moore(pos, radius, get_agents, include_empty)
+        else:
+            neighbors = self._von_neumann(
+                pos, radius, get_agents, include_empty)
 
-        """
-        neighborhood = self.iter_neighborhood(
-            pos, moore, include_center, radius)
-        return self.iter_cell_list_contents(neighborhood)
+        return neighbors
 
-    def get_neighbors(self, pos, moore,
-                      include_center=False, radius=1):
-        """ Return a list of neighbors to a certain point.
+    def at_row(self, row, include_agents=False):
```

Comment:
Not sure about the function name. What does the "at" refer to? (for me it sound like it would return a boolean)
Maybe name it row_iter (as coord_iter function, but am not happy with that name either)